client-app/app/components/panel-resizer.js

```diff
@@ -1,74 +1,116 @@
 import Component from "@ember/component";
+import { scheduleOnce, throttle } from "@ember/runloop";
+
 const MOVE_EVENTS = ["touchmove", "mousemove"];
 const UP_EVENTS = ["touchend", "mouseup"];
 const DOWN_EVENTS = ["touchstart", "mousedown"];
 
+function bind(target, key, desc) {
```

Comment:
This decorator is a small experiment I did that I'd like someone to review. It solves the problem where we have a component method used as an event handler that we need to later remove when the component is destroyed, and we want `this` inside the method to refer to the component and not the DOM element the event is attached to.

Currently I'm doing this:

```js
endDrag() {
  if (this._endDrag) return this._endDrag;

  const handler = () => {
    this.set("clicked", true);
    console.log("Element clicked!");
  };
  this.set("_endDrag", handler);
  return this._endDrag;
}
```

and in `didInsertElement`:

```js
document.addEventListener("click", this.endDrag());
```

and in `willDestroyElement`:

```js
document.removeEventListener("click", this.endDrag());
```

We have a few more event handlers in this component so this can get a bit repetitive.

This decorator allows us to do something like this:

```js
@bind
endDrag() {
  this.set("clicked", true);
  console.log("Element clicked!");
}
// later
document.addEventListener("click", this.endDrag);
document.removeEventListener("click", this.endDrag);
```

Any objections to this?

Reply:
I am probably not the best person to review this perhaps @jjaffeux or @CvX  can have a look (and at the entire change) ... overall I am super happy with all of this